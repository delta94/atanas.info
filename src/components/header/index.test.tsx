/* eslint-disable @typescript-eslint/ban-ts-comment */

import * as React from 'react';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';

import { Header } from '.';

describe('Header component', () => {
	it('Should render the Header component', () => {
		const wrapper = mount(<Header />);

		expect(wrapper).toMatchSnapshot();

		// codebeat:disable[ABC]
		act(() => {
			// @ts-ignore
			wrapper.find('.c-nav a').at(0).props().onClick();

			expect(wrapper).toMatchSnapshot();

			// @ts-ignore
			wrapper.find('button').at(0).props().onClick();

			expect(wrapper).toMatchSnapshot();
		});
		// codebeat:enable[ABC]
	});
});
