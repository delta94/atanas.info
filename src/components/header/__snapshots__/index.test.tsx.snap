// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render the Header component 1`] = `
<Component>
  <header
    className="c-header"
  >
    <div
      className="o-shell o-shell--flex"
    >
      <a
        className="c-logo"
        href="#hello"
        title="Back to top"
      >
        <Component
          className="c-svg-logo"
          name="svg-logo"
        >
          <svg
            className="c-svg-logo"
          >
            <use
              xlinkHref="#svg-logo"
            />
          </svg>
        </Component>
      </a>
      <button
        className="c-nav__toggle"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </button>
      <nav
        className="c-nav"
      >
        <ul>
          <li
            key="0"
          >
            <a
              href="#about"
              onClick={[Function]}
              title="About me"
            >
              About
            </a>
          </li>
          <li
            key="1"
          >
            <a
              href="#skills"
              onClick={[Function]}
              title="My skills"
            >
              Skills
            </a>
          </li>
          <li
            key="2"
          >
            <a
              href="#stats"
              onClick={[Function]}
              title="My stats"
            >
              Stats
            </a>
          </li>
          <li
            key="3"
          >
            <a
              href="#portfolio"
              onClick={[Function]}
              title="My portfolio"
            >
              Portfolio
            </a>
          </li>
          <li
            key="4"
          >
            <a
              href="#slides"
              onClick={[Function]}
              title="My presentations"
            >
              Slides
            </a>
          </li>
          <li
            key="5"
          >
            <a
              href="#videos"
              onClick={[Function]}
              title="My video presentations"
            >
              Videos
            </a>
          </li>
          <li
            key="6"
          >
            <a
              href="./resume.pdf"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="My CV"
            >
              Resume (pdf)
            </a>
          </li>
          <li
            key="7"
          >
            <a
              href="projects/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="My open source projects"
            >
              Open source
            </a>
          </li>
          <li
            key="8"
          >
            <a
              href="https://at-the-wall.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="At the Wall - A Game of Thrones scene"
            >
              At the Wall
            </a>
          </li>
          <li
            key="9"
          >
            <a
              href="https://scriptex.js.org/2048/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Join the numbers and get to the 2048 tile!"
            >
              2048
            </a>
          </li>
          <li
            key="10"
          >
            <a
              href="https://material-tetris.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Material Tetris. Play here!"
            >
              Material Tetris
            </a>
          </li>
          <li
            key="11"
          >
            <a
              href="https://material-snake.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Material Snake. Play here!"
            >
              Material Snake
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</Component>
`;

exports[`Header component Should render the Header component 2`] = `
<Component>
  <header
    className="c-header"
  >
    <div
      className="o-shell o-shell--flex"
    >
      <a
        className="c-logo"
        href="#hello"
        title="Back to top"
      >
        <Component
          className="c-svg-logo"
          name="svg-logo"
        >
          <svg
            className="c-svg-logo"
          >
            <use
              xlinkHref="#svg-logo"
            />
          </svg>
        </Component>
      </a>
      <button
        className="c-nav__toggle"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </button>
      <nav
        className="c-nav"
      >
        <ul>
          <li
            key="0"
          >
            <a
              href="#about"
              onClick={[Function]}
              title="About me"
            >
              About
            </a>
          </li>
          <li
            key="1"
          >
            <a
              href="#skills"
              onClick={[Function]}
              title="My skills"
            >
              Skills
            </a>
          </li>
          <li
            key="2"
          >
            <a
              href="#stats"
              onClick={[Function]}
              title="My stats"
            >
              Stats
            </a>
          </li>
          <li
            key="3"
          >
            <a
              href="#portfolio"
              onClick={[Function]}
              title="My portfolio"
            >
              Portfolio
            </a>
          </li>
          <li
            key="4"
          >
            <a
              href="#slides"
              onClick={[Function]}
              title="My presentations"
            >
              Slides
            </a>
          </li>
          <li
            key="5"
          >
            <a
              href="#videos"
              onClick={[Function]}
              title="My video presentations"
            >
              Videos
            </a>
          </li>
          <li
            key="6"
          >
            <a
              href="./resume.pdf"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="My CV"
            >
              Resume (pdf)
            </a>
          </li>
          <li
            key="7"
          >
            <a
              href="projects/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="My open source projects"
            >
              Open source
            </a>
          </li>
          <li
            key="8"
          >
            <a
              href="https://at-the-wall.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="At the Wall - A Game of Thrones scene"
            >
              At the Wall
            </a>
          </li>
          <li
            key="9"
          >
            <a
              href="https://scriptex.js.org/2048/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Join the numbers and get to the 2048 tile!"
            >
              2048
            </a>
          </li>
          <li
            key="10"
          >
            <a
              href="https://material-tetris.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Material Tetris. Play here!"
            >
              Material Tetris
            </a>
          </li>
          <li
            key="11"
          >
            <a
              href="https://material-snake.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Material Snake. Play here!"
            >
              Material Snake
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</Component>
`;

exports[`Header component Should render the Header component 3`] = `
<Component>
  <header
    className="c-header"
  >
    <div
      className="o-shell o-shell--flex"
    >
      <a
        className="c-logo"
        href="#hello"
        title="Back to top"
      >
        <Component
          className="c-svg-logo"
          name="svg-logo"
        >
          <svg
            className="c-svg-logo"
          >
            <use
              xlinkHref="#svg-logo"
            />
          </svg>
        </Component>
      </a>
      <button
        className="c-nav__toggle"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </button>
      <nav
        className="c-nav"
      >
        <ul>
          <li
            key="0"
          >
            <a
              href="#about"
              onClick={[Function]}
              title="About me"
            >
              About
            </a>
          </li>
          <li
            key="1"
          >
            <a
              href="#skills"
              onClick={[Function]}
              title="My skills"
            >
              Skills
            </a>
          </li>
          <li
            key="2"
          >
            <a
              href="#stats"
              onClick={[Function]}
              title="My stats"
            >
              Stats
            </a>
          </li>
          <li
            key="3"
          >
            <a
              href="#portfolio"
              onClick={[Function]}
              title="My portfolio"
            >
              Portfolio
            </a>
          </li>
          <li
            key="4"
          >
            <a
              href="#slides"
              onClick={[Function]}
              title="My presentations"
            >
              Slides
            </a>
          </li>
          <li
            key="5"
          >
            <a
              href="#videos"
              onClick={[Function]}
              title="My video presentations"
            >
              Videos
            </a>
          </li>
          <li
            key="6"
          >
            <a
              href="./resume.pdf"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="My CV"
            >
              Resume (pdf)
            </a>
          </li>
          <li
            key="7"
          >
            <a
              href="projects/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="My open source projects"
            >
              Open source
            </a>
          </li>
          <li
            key="8"
          >
            <a
              href="https://at-the-wall.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="At the Wall - A Game of Thrones scene"
            >
              At the Wall
            </a>
          </li>
          <li
            key="9"
          >
            <a
              href="https://scriptex.js.org/2048/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Join the numbers and get to the 2048 tile!"
            >
              2048
            </a>
          </li>
          <li
            key="10"
          >
            <a
              href="https://material-tetris.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Material Tetris. Play here!"
            >
              Material Tetris
            </a>
          </li>
          <li
            key="11"
          >
            <a
              href="https://material-snake.atanas.info/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Material Snake. Play here!"
            >
              Material Snake
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</Component>
`;
